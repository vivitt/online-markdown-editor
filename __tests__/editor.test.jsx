import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { afterEach, describe, it, expect } from "vitest";
import userEvent from "@testing-library/user-event";

import Page from "../app/editor/page";

describe("Editor view tests:", () => {
  const user = userEvent.setup();
  afterEach(cleanup);

  it("renders editor view", () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  it("renders edit panel", () => {
    render(<Page />);
    const editSection = screen.getByText("Edit");
    expect(editSection).toBeDefined();
  });

  it("renders preview panel", () => {
    render(<Page />);

    const previewSection = screen.getByText("Preview");
    expect(previewSection).toBeDefined();
  });

  it("renders a textArea", () => {
    render(<Page />);
    const textarea = screen.getByRole("textbox", { lang: "markdown" });
    expect(textarea).toBeDefined();
  });

  it("textarea value change when user type", async () => {
    render(<Page />);
    const input = screen.getByRole("textbox");
    input.focus()
    await user.keyboard("Good day");
    expect(input).toHaveProperty("value", "Good day");
  });

  it("preview area value match edit area value", async () => {
    render(<Page />);
    const input = screen.getByRole("textbox");
    input.focus()
    await user.keyboard("hola");
    const preview = screen.getByText("hola");
    expect(preview).toBeDefined();
  });

  it("doesn't add numbers when input is empty", async () => {
    render(<Page />);
    
    expect(screen.queryAllByText("1")).toHaveLength(0)
  });

  it("adds a line number when user type", async () => {
    render(<Page />);
    const input = screen.getByRole("textbox");
    input.focus()
    await user.keyboard("hola");
    expect(screen.queryAllByText("1")).toHaveLength(1)
  });

  it("add a number if enter key is pressed in the texarea", async () => {
    render(<Page />);
    const input = screen.getByRole("textbox");
    input.focus()
    await user.keyboard('[Enter]');
    expect(screen.getAllByText("1")).toHaveLength(1)
    expect(screen.queryAllByText("2")).toHaveLength(0)
  });
});
