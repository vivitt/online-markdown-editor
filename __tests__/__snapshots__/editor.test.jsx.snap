// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Editor view tests: > renders editor view 1`] = `
<div>
  <div
    class="md:flex flex-wrap relative w-full"
  >
    <div
      class="w-3/6"
    >
      <button
        class="border-r border-t border-l rounded-sm p-2"
      >
         
        Edit
         
      </button>
      <div
        class="border"
      >
        <div
          class="flex  font-mono"
        >
          <div
            aria-hidden="true"
            class="w-5 flex flex-col items-center px-5 "
            id="line-numbers"
          />
          <textarea
            aria-autocomplete="list"
            class="resize-none w-full h-screen focus:outline-0"
            data-language="markdown"
            id="edit"
          />
        </div>
      </div>
    </div>
    <div
      class="w-3/6"
    >
      <button
        class="border-r border-t border-l rounded-sm p-2"
      >
         
        Preview
         
      </button>
      <div
        class="border"
      >
        <div
          class="prose px-3 w-full bg-white  h-screen"
        />
      </div>
    </div>
  </div>
</div>
`;
